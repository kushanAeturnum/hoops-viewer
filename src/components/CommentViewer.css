.comment-viewer{
    padding: 5%;
    text-align: center;
}

