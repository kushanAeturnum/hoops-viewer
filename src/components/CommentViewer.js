import React, { useEffect,useState } from 'react';
import { Card } from 'react-bootstrap';
import './CommentViewer.css';

const CommentViewer = (props) => {

    const [data, setData] = useState([])
    var token = 'eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ3b29kd2FyZGJzbXVzZXIxIiwiZXhwIjoxNjQwODczMjU3LCJpYXQiOjE2NDA4MzcyNTd9.a4FAnLPKQm3o0-QJufSa_jPFem1zN-lxRcqM3P7Yg-s';
    var prequote_rfq_id = "61c19377e13c01425d1c3a62";
    var comment_channel_id = "ef8fa86a-6239-11ec-b78e-e9b71b80d45f";
    useEffect(() => {
     fetch(`http://apriordevapp-env.eba-ajmpa4mr.us-east-1.elasticbeanstalk.com/api/ws/comment-threads/prequote-rfqs/${prequote_rfq_id}/comment-channels/${comment_channel_id}`, {
            method: 'GET',
            //body: JSON.stringify(body), // string or object
            headers: {
              'Content-Type': 'application/json',
              'Authorization': `Bearer ${token}`
            }
          }).then(response => response.json())
          .then(data => {
            console.log('Success:', data);
            setData(data.data);
          })
          .catch((error) => {
            console.error('Error:', error);
          });

    }, []);

    // const commentClicker = () =>{
    //    let commentId = 1;
    //    props.clickedComment(commentId)
    // }
    return (
        <div className='comment-viewer'>
            <h1 className='comment-text'>Comments History</h1>
           <div className='comment-thred'>
           {data && data.map((commentThread, index) => 
           <Card 
           key={index}
           onClick={()=> {props.clickedComment(commentThread.id)}}
      style={{ width: '100%', borderRadius: '10px', padding: '20px', marginBottom: '10px' }}
      className="comment_thread_card shadow"
    >
      <div  style={{marginBottom: '10px', fontWeight: 'bold'}} className="commentTitle">{commentThread.commentTitle}</div>
      <div className="createdTime">{commentThread.createdTime}</div>
     
    </Card>
        
           
           ) } 
           </div>
           
        </div>
    )
}

export default CommentViewer
