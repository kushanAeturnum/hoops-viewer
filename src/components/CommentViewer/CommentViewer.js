import React, { useEffect,useState } from 'react';
import { Card } from 'react-bootstrap';
import './CommentViewer.css';

const CommentViewer = (props) => {

    const [data, setData] = useState([])
    var token = 'eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ3b29kd2FyZGJzbXVzZXIxIiwiZXhwIjoxNjQxMzMyMzAxLCJpYXQiOjE2NDEyOTYzMDF9.2-WsXwnDx-YfzXzfUCaPYFptu55WREYwDEyRjxyCL_U';
    var prequote_rfq_id = "61c96c51e13c01425d1c3a73";
    var comment_channel_id = "4dba5c05-66e7-11ec-bb56-777a0c3ed608";
    useEffect(() => {
     fetch(`http://apriordevapp-env.eba-ajmpa4mr.us-east-1.elasticbeanstalk.com/api/ws/comment-threads/prequote-rfqs/${prequote_rfq_id}/comment-channels/${comment_channel_id}`, {
            method: 'GET',
            headers: {
              'Content-Type': 'application/json',
              'Authorization': `Bearer ${token}`
            }
          }).then(response => response.json())
          .then(data => {
            console.log('Success:', data);
            setData(data.data);
          })
          .catch((error) => {
            console.error('Error:', error);
          });

    }, []);

    return (
        <div className='comment-viewer'>
           <h1 className='comment-text' data-testid="commentText">Comments History</h1>
           <div className='comment-thred'>
           {data && data.map((commentThread, index) => 
            <Card 
              key={index}
              onClick={()=> {props.clickedComment(commentThread.id)}}
                style={{ width: '100%', borderRadius: '10px', padding: '20px', marginBottom: '10px' }}
                className="comment_thread_card shadow"
              >
                <div  style={{marginBottom: '10px', fontWeight: 'bold'}} className="commentTitle">{commentThread.commentTitle}</div>
                <div className="createdTime">{commentThread.createdTime}</div>
            </Card>
           )} 
           </div>
           
        </div>
    )
}

export default CommentViewer
