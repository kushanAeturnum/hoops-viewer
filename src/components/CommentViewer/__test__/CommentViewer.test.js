import React from "react";
import reactDom from "react-dom";
import CommentViewer from "../CommentViewer";

import {render, cleanup, screen} from '@testing-library/react';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('match snapshot', () => {
    const tree = renderer.create(<CommentViewer/>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('comment header', () => {
    render(<CommentViewer/>);
    screen.getByText("Comments History");
})

