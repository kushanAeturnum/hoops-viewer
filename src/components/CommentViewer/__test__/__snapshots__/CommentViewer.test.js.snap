// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="comment-viewer"
>
  <h1
    className="comment-text"
    data-testid="commentText"
  >
    Comments History
  </h1>
  <div
    className="comment-thred"
  />
</div>
`;
